### DISCUSSION TITLE
Disconnected?

### DISCUSSION THREAD

**Discussion by drgkt** [arendst/Tasmota#12740] [Q&A] [answered]
I issued an http ON command only to discover it failed.
Looked in router and sure enough the device was nowhere to be found.
I have configured a static IP for the switch. I unplugged it and plugged it in again and it works again.
Funny thing was that the blue light was not flashing to indicate it got disconnected from the wifi either.

What might have happened and how to prevent it from happening again?
Lastly, how might the blue light behavior be explained?

**Comment by drgkt** [comment]
Thank you.
However, this certainly spoils the tasmota fame. I have here WeMo's and eWeLink products that were never found off-line. I just pinged all 2 tasmota plugs I have connected to my wifi and there is no answer. Meanwhile the flashing blue light indicating to wifi connection is NOT present in neither. The only solution is to unplug / replug to mains. :-(

**Reply by barbudor** [reply]
There are many possible reason that a specific device is less reliable than others.

It could be an hardware design problems. If you don't have same devices running with Tasmota and WeMo/eWeLink to compare, it is difficult to conclude. May be that device was also unreliable with it's firmware.

It could be an hardware design with limited power supply capability. Due to it's large feature set, Tasmota is known to be slightly more power-hungry than original manufacturers firmware. So if the device's power-supply is designed on the cheap, it may be less reliable with Tasmota. Commands like [`Sleep`](https://tasmota.github.io/docs/Commands/#sleep) may help tuning Tasmota power need.

Some network environment tends not be the best. Tasmota has implemented features such as [`SetOption41`](https://tasmota.github.io/docs/Commands/#setoption41) to overcome those problems. It is disabled by default but you may require it on your network. Other firmware may have decided to enforce such feature in all cases even if not needed.

**Comment by drgkt** [comment]
THANK YOU!
That makes lots of sense!
I did not measure the exact time frame that devices go offline but if I had to guess it must be  3-4 weeks.
Why when they go off line there is no blue light blinking?
20:42:04 CMD: sleep
20:42:04 RSL: /RESULT = {"Sleep":{"50":{"Active":"50"}}}
20:42:54 CMD: setoption41
20:42:54 RSL: /RESULT = {"SetOption41":0}

What would you recommend these settings be?

**Reply by barbudor** [reply]
How would I know what can work in your system without having a single clue on what are your devices and the root cause.
It's up to you now to do some experimentation and analysis.

However, for `SetOption41`, I would test with 60 seconds.
And for `Sleep`, I would use 100 (double of default 50).
But not both at the same time on the same device.

**Comment by drgkt** [comment, marked as answer]
Hmm...
If you did not answer I was gonna put zero for sleep (disable it) and 160 (every hour) for setoption41, both at the same time on the same device.

Why "both"is not a good idea?

I looked back and saw that the last time they were unreachable was 24 days ago.
FYI the devices are preflashed Athom FW 9.1.0

For academic reasons, why when they go off-line the blue light does not start flashing?

**Reply by barbudor** [reply]
> If you did not answer I was gonna put zero for sleep (disable it) and 

If the problem is Tasmota being too power hungry, the solution would be to get it sleep longer in order to reduce power consumption.
Disabling the Sleep mode would not help.
If you want a more aggressive test, make it 250 but that could be at the expense of reactivity.
Alternatively, disabling the sleep mode to increase power consumption could be a means to see if it makes the problem more obvious than once every 3 weeks which is also a way to diagnose it.

> 160 (every hour) for setoption41, both at the same time on the same device.

Gratutious ARP are a way too tell the network "Hey don't forget me I'm still here" and workaround timeouts in network equipment. Without a single clue if this is the problem and what could be the timeout in your equipment (you may have no way to know it they are not manageable, professional devices), I would have started by 60 seconds but 5 minutes could be good too, 1 hour could work too.

> Why "both"is not a good idea?

Because if you take 2 medicines at the same time, you don't know whch one is the cure, and addition of both could be worse.
When debugging things, always test changes once by one.

> I looked back and saw that the last time they were unreachable was 24 days ago.
> FYI the devices are preflashed Athom FW 9.1.0

> For academic reasons, why when they go off-line the blue light does not start flashing?

Crystal ball guess ?
If it is a power supply problem, then may be the device is crashed. 
I don't know neither what you have tried to diagnose when the problem occur.
Is that only Web GUI that doesn't work ?
Does local control (like the push button on a Sonoff Basic) still works ?
Does MQTT still works ? What are the logs of your MQTT broker ? Did it detected a loss of connection ?
...

I game you some possible directions but if you don't investigate yourself, there's nothing I can do from here.

**Comment by drgkt** [comment]
Sorry for the delay.

The Athom smart plugs we are talking about here are only used in local LAN to turn on/off after http://xxx.xxx.xxx.xxx/cm?cmnd=power%20on(ff). No MQTT is associated with them.

Yes, only this does not work when connection is lost (ping and web interface also of course), and the blue light (which was TURNED OFF via Template configuration GPIO13: Ledlink_i), does not blink.

The device BUTTON still works which per the logic explained indicates that no crash is involved.

I spent some time reading about SLEEP, dynamic and normal and their options. I admit, most of it went over my head.

So as far as SLEEP, I am thinking, why put it to sleep at all? How much power will a smart plug internal circuitry can possibly consume?
Will ever something go wrong, if I set SLEEP to zero (disable it)?

I understand why changing more than one parameter at once makes troubleshooting difficult.

Perhaps all this issue needs is "option41". Will let you know in a month!
I left SLEEP to default.

Now, why the blue light does not flash to broadcast "Hey! I lost connection!" who knows?

**Comment by drgkt** [comment]
Well, it has been 9 days now and they are both off line.
I noticed that when I press the plug's button it momentarily flashes blue, then turns to red and arms the relay. When they are online, pressing the button just turns red.
Even after arming disarming relay, device is offline.

I will put back option41 to default and disable sleep.
I will report back.

@barbudor
I appreciate your input, without which I would just have to ditch them as unreliable!

**Comment by drgkt** [comment]
Update:
One device has problem with ping. Seldomlly returns all 4 packets. That is probably why the http on-off command sometimes fails.
Signal (68%, -66 dBm) 11n (not further than 5 meters from router...

**Comment by drgkt** [comment]
1 device in which Sleep was set to 250 suddenly yesterday crashed in the ON power state. The device became NON responsive to its button.

I cannot help it, I must say again that I did not expect these problems on Tasmota. I have eWayLink and Wemo devices that to date have not failed.

**Comment by drgkt** [comment]
Pinging xxx.xxx.xxx.xxx with 32 bytes of data:

Reply from : bytes=32 time=141ms TTL=255
Reply from : bytes=32 time=48ms TTL=255
Request timed out.
Reply from : bytes=32 time=60ms TTL=255

Ping statistics for 10.206.103.14:
    Packets: Sent = 4, Received = 3, Lost = 1 (25% loss),
Approximate round trip times in milli-seconds:
    Minimum = 48ms, Maximum = 141ms, Average = 83ms

**Comment by drgkt** [comment]
Can someone list some non tasmota wifi smart outlets / plugs that can be controlled (on-off) from a browser (locally) but also accept a command like http://xxx.xxx.xxx.xxx/cm?cmnd=power%20off or on?
